frontend/packages/dev-console/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx

```diff
@@ -1,30 +1,29 @@
 import * as React from 'react';
-import * as _ from 'lodash';
 import { connect } from 'react-redux';
 import { impersonateStateToProps } from '@console/internal/reducers/ui';
-import { K8sResourceKind } from '@console/internal/module/k8s';
 import { useAccessReview } from '@console/internal/components/utils';
 import { Button } from 'patternfly-react';
 import { rerunPipeline } from '../../../utils/pipeline-actions';
 import { PipelineModel } from '../../../models';
+import { getLatestRun, PipelineRun } from '../../../utils/pipeline-augment';
 
 type TriggerLastRunButtonProps = {
-  pipeline: K8sResourceKind;
-  disabled?: boolean;
+  pipelineRuns: PipelineRun[];
   impersonate?;
 };
 
 const TriggerLastRunButton: React.FC<TriggerLastRunButtonProps> = ({
-  pipeline,
+  pipelineRuns,
   impersonate,
-  disabled,
 }) => {
-  const latestRun = _.get(pipeline, ['latestRun'], null);
-  const { label, callback, accessReview } = rerunPipeline(PipelineModel, pipeline, latestRun);
+  const latestRun = getLatestRun({ data: pipelineRuns }, 'startTimestamp');
+  const { label, callback, accessReview } = rerunPipeline(PipelineModel, latestRun, null, {
+    pipelineRunRedirect: false,
+  });
   const isAllowed = useAccessReview(accessReview, impersonate);
```

Comment:
This probably shouldn't be a hook. `accessReview` is optional but it *must* call this function every time due to the way hooks work.